sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/ReplicatedResourceClient.java

```diff
@@ -171,6 +167,81 @@ public Mono<StoreResponse> invokeAsync(RxDocumentServiceRequest request,
             addressSelector);
     }
 
+    private Mono<StoreResponse> getStoreResponseMonoWithSpeculation(RxDocumentServiceRequest request, Quadruple<Boolean, Boolean, Duration, Integer> forceRefreshAndTimeout) {
+        CosmosE2EOperationRetryPolicyConfig config = request.requestContext.getE2EOperationRetryPolicyConfig();
+        List<Mono<StoreResponse>> monoList = new ArrayList<>();
+        List<RxDocumentServiceRequest> requests = new ArrayList<>();
+
+        if (speculativeProcessor.getRegionsForPureExploration() != null
+            && !speculativeProcessor.getRegionsForPureExploration().isEmpty()) {
+            explore(request, forceRefreshAndTimeout);
```

Comment:
can explore also be part of the Mono list ?I guess no because we do not want to cancel them (firstWithValue)